### ISSUE TITLE
Allow realms to integrate with an SMTP server

### ISSUE THREAD

**Issue by whaught** [google/exposure-notifications-verification-server#787]
### TL;DR
Currently we use the default firebase email flow for password reset.
If we wish to use email for a 2nd factor, controler/eliminate rate-limiting concerns, and customize invitations, we need to be able to send our own emails.

### Design

**Proposal**

1. Use the golang SMTP package to send email from a custom template
2. Include a realm setting for integrating with an email server credential
3. Create a realm setting that includes a text template for inviting new users.
4. Replace our hacked SendVerificationEmail with the template and use the server SDK generate link (which has 10x quota).

**Alternatives considered**
Integrate directly with SendGrid APIs

**Additional information**
Once we have this capability we can consider using this for TOTP sent via email for 2nd factor auth along with Twilio to replace the SMS based auth.
